core/domain/entities/routing/data_nodes/core/CurrentUser.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace EventEspresso\core\domain\entities\routing\data_nodes\core;
+
+use EventEspresso\core\services\json\JsonDataNode;
+use EventEspresso\core\services\json\JsonDataNodeValidator;
+use WP_User;
+
+/**
+ * Class CurrentUser
+ *
+ * @package EventEspresso\core\domain\entities\routing\data_nodes
+ * @author  Brent Christensen
+ * @since   $VID:$
+ */
+class CurrentUser extends JsonDataNode
+{
+
+    const NODE_NAME = 'currentUser';
+
+
+    /**
+     * JsonDataNodeHandler constructor.
+     *
+     * @param JsonDataNodeValidator $validator
+     */
+    public function __construct(JsonDataNodeValidator $validator)
+    {
+        parent::__construct($validator);
+        $this->setNodeName(CurrentUser::NODE_NAME);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function initialize()
+    {
+        $current_user = wp_get_current_user();
+        if (! $current_user instanceof WP_User) {
+            $current_user = new WP_User();
+        }
+        $this->addData('description', $current_user->description);
+        $this->addData('email', $current_user->email);
+        $this->addData('firstName', $current_user->firstName);
+        $this->addData('name', $current_user->name);
+        $this->addData('nicename', $current_user->nicename);
+        $this->addData('nickname', $current_user->nickname);
+        $this->addData('lastName', $current_user->lastName);
+        $this->addData('locale', $current_user->locale);
+        $this->addData('userId', $current_user->userId);
+        $this->addData('username', $current_user->username);
+        $this->addData('__typename', 'User');
```

Comment:
No guid here. you need to add `id` and replace deprecated `userId` with `databaseId`

Reply:
Also, the values for all the fields (like `email`, `locale` etc.) are missing

![image](https://user-images.githubusercontent.com/18226415/86580973-0dc4e680-bf9d-11ea-83e5-093b7a21a0e4.png)

Reply:
```suggestion
        $this->addData('id', Relay::toGlobalId('user', $current_user->ID));
        $this->addData('description', $current_user->description);
        $this->addData('email', $current_user->user_email);
        $this->addData('firstName', $current_user->first_name);
        $this->addData('name', $current_user->display_name);
        $this->addData('nicename', $current_user->user_nicename);
        $this->addData('nickname', $current_user->nickname);
        $this->addData('lastName', $current_user->last_lame);
        $this->addData('locale', get_user_locale($current_user));
        $this->addData('databaseId', $current_user->ID);
        $this->addData('username', $current_user->user_login);
        $this->addData('__typename', 'User');
```